controller/cache/cache.go

```diff
@@ -295,9 +305,17 @@ func (c *liveStateCache) getCluster(server string) (clustercache.ClusterCache, e
 			res := &ResourceInfo{}
 			populateNodeInfo(un, res)
 			res.Health, _ = health.GetResourceHealth(un, cacheSettings.clusterSettings.ResourceHealthOverride)
-			appName := kube.GetAppInstanceLabel(un, cacheSettings.appInstanceLabelKey)
-			if isRoot && appName != "" {
-				res.AppName = appName
+
+			if trackingMethod == "annotation" {
```

Comment:
Two suggestions here - since we use at least 3 options ("label", "annotation" or "annotation+label"), it might make sense here to use a `switch` instead of `if/else` branch and also use constants instead of string literals for the values

Reply:
Or maybe even two distinct methods, that encapsulate all the logic behind it, something like

`GetAppIdentifier(target *unstructured.Unstructured, key, appIdMethod string)`

and

`SetAppIdentifier(target *unstructured.Unstructured, key, val, appIdMethod string)`

Or even encapsulate the `appIdType` into it's own type, such as

```go
type ApplicationIdentifierMethod string
const (
  AppIdMethodAnnotation  ApplicationIdentifierMethod = "annotation"
  AppIdMethodLabel ApplicationIdentifierMethod = "label"
  AppIdMethodAnnotationAndLabel ApplicationIdentifierMethod = "annotation+label"
)
```
This would ensure only good values are passed to the respective methods.

Reply:
Hey @jannfis . Thank you for review. It is not ready code, I just pushed my progress. But thank you for comments I will take it into account